File: src/main/java/renankummer/udemy/springwebapp/model/Publisher.java
package renankummer.udemy.springwebapp.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Objects;
import java.util.Set;

@Entity
@Data @NoArgsConstructor @AllArgsConstructor @Builder
public class Publisher {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "name", nullable = false)
    private String name;
    @Column(name = "address_line1", nullable = false)
    private String addressLine1;
    @Column(name = "state", nullable = false)
    private String state;
    @Column(name = "city", nullable = false)
    private String city;
    @Column(name = "zip_code", nullable = false)
    private String zipCode;

    @OneToMany
    @JoinColumn(name = "publisher_id")
    @Builder.Default
    private Set<Book> books = new HashSet<>();

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Publisher publisher = (Publisher) o;
        return Objects.equals(id, publisher.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: src/main/kotlin/renankummer/udemy/springwebapp/models/Publisher.kt
package renankummer.udemy.springwebapp.models

import javax.persistence.*

@Entity
data class Publisher(
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        val id: Long = 0,

        @Column(name = "name", nullable = false)
        var name: String = "",
        @Column(name = "addressLine1", nullable = false)
        var addressLine1: String = "",
        @Column(name = "city", nullable = false)
        var city: String = "",
        @Column(name = "state", nullable = false)
        var state: String = "",
        @Column(name = "zipCode", nullable = false)
        var zipCode: String = "",

        @OneToMany
        @JoinColumn(name = "publisher_id")
        var books: MutableSet<Book> = mutableSetOf()
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Publisher

        if (id != other.id) return false

        return true
    }

    override fun hashCode() = id.hashCode()
}
